virtualizarr/kerchunk.py

```diff
@@ -223,7 +224,7 @@ def dataset_to_kerchunk_refs(ds: xr.Dataset) -> KerchunkStoreRefs:
 
     all_arr_refs = {}
     for var_name, var in ds.variables.items():
-        arr_refs = variable_to_kerchunk_arr_refs(var, var_name)
+        arr_refs = variable_to_kerchunk_arr_refs(var, str(var_name))
```

Comment:
All of these `str` conversions need to be looked at more closely. We do a lot of stuff with `Dataset.variables` / `.items` / `.coords` / etc. In xarray, `Variable.name` is just `Hashable`. Lots of types in virtualzarr are `str` (which is stricter, since `str` is hashable, but not all `Hashable` are str).


We might be able to loosen this a bit, but there's probably no benefit to.

Reply:
I'm not sure we really need to worry about this. It's a little used feature in xarray at all, and if netCDF / zarr groups don't support `Hashable` then it won't be relevant for VirtualiZarr.